src/server/services/RepositoryService.ts

```diff
@@ -527,93 +544,40 @@ export default class RepositoryService implements IRepositoryService {
     }
   }
 
-  private storeMetadata(
+  private storeJSON(
     matchQuality: MatchQuality,
     chainId: string,
     address: string,
-    metadata: Metadata
+    fileName: string,
+    contentJSON: any
   ) {
     this.save(
       {
         matchQuality,
         chainId,
         address,
-        fileName: "metadata.json",
-      },
-      JSON.stringify(metadata)
-    );
-  }
-
-  private storeConstructorArgs(
-    matchQuality: MatchQuality,
-    chainId: string,
-    address: string,
-    abiEncodedConstructorArguments: string
-  ) {
-    this.save(
-      {
-        matchQuality,
-        chainId,
-        address,
-        source: false,
-        fileName: "constructor-args.txt",
-      },
-      abiEncodedConstructorArguments
-    );
-  }
-
-  private storeContextVariables(
-    matchQuality: MatchQuality,
-    chainId: string,
-    address: string,
-    contextVariables: ContextVariables
-  ) {
-    this.save(
-      {
-        matchQuality,
-        chainId,
-        address,
-        source: false,
-        fileName: "context-variables.json",
-      },
-      JSON.stringify(contextVariables, undefined, 2)
-    );
-  }
-
-  private storeCreate2Args(
-    matchQuality: MatchQuality,
-    chainId: string,
-    address: string,
-    create2Args: Create2Args
-  ) {
-    this.save(
-      {
-        matchQuality,
-        chainId,
-        address,
-        source: false,
-        fileName: "create2-args.json",
+        fileName,
       },
-      JSON.stringify(create2Args)
+      JSON.stringify(contentJSON)
```

Comment:
I just noticed that `JSON.stringify(contextVariables, undefined, 2)` is not used. I'm personally ok with that, but there is less readability for a human manually reading the repository. If you also think this is not a problem no need to change it.

Reply:
Yes I haven't added it here and removed it in other places to keep consistency https://github.com/ethereum/sourcify/pull/942/commits/d1034319fdc5fa8fa76e25136e77719a93ad06f7 the metadata.json also does not have spaces. Apart from metadata, the JSON files are not big and would be still readable.